contrib/externs/chrome_extensions.js

```diff
@@ -11214,3 +11214,80 @@ chrome.downloads.DeterminingFilenameEvent.prototype.hasListeners =
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/** @type {string} */
+chrome.declarativeContent.PageStateMatcherInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.ShowPageActionInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.SetIconInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.RequestContentScriptInstanceType;
+
+
+/**
+ * @typedef {{
+ *   pageUrl: (!UrlFilter|undefined),
+ *   css: (!Array<string>|undefined),
+ *   isBookmarked: (boolean|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcherLiteral;
+
+
+/**
+ * @constructor
+ * @param {?chrome.declarativeContent.PageStateMatcherLiteral=} literalValue
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcher = function(literalValue) {};
+
+
+/**
+ * @constructor
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-ShowPageAction
+ */
+chrome.declarativeContent.ShowPageAction = function() {};
+
+
+/**
+ * @typedef {{
+ *   imageData: (!ImageData|!Object<number, !ImageData>|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-SetIcon
+ */
+chrome.declarativeContent.SetIcon;
```

Comment:
Do you expect to have code that does `new SetIcon()`?
It's not clear to me from the doc reference.

Reply:
Never used it, but from https://developer.chrome.com/extensions/declarativeContent#rules:

> As a declarative API, this API lets you register rules on the onPageChanged event object which take an action (ShowPageAction and SetIcon)....

It could be deduced that it can be expected (since `new ShowPageAction()` is used).

Reply:
If you need to call `new SetIcon()`, then `SetIcon` must be defined as `@constructor` rather than `@typedef`.